airflow/utils/dag_processing.py

```diff
@@ -1229,6 +1243,20 @@ def heartbeat(self):
 
         return simple_dags
 
+    def _kill_timed_out_processors(self):
+        """
+        Kill any file processors that timeout to defend against process hangs.
+        """
+        now = timezone.utcnow()
+        for file_path, processor in self._processors.items():
+            duration = now - processor.start_time
+            if duration > self._processor_timeout:
+                self.log.info(
+                    "Processor for %s with PID %s started at %s has timed out, "
+                    "killing it.",
+                    processor.file_path, processor.pid, processor.start_time.isoformat())
+                processor.kill()
```

Comment:
Lets add a metric for number of parsing processes killed.

Reply:
Agreed, will add.